src/internal/connector/onedrive/restore.go

```diff
@@ -696,26 +797,55 @@ func restoreData(
 		return "", details.ItemInfo{}, err
 	}
 
-	// Get a drive item writer
-	w, err := driveItemWriter(ctx, service, driveID, ptr.Val(newItem.GetId()), ss.Size())
-	if err != nil {
-		return "", details.ItemInfo{}, err
-	}
+	var (
+		w       io.Writer
+		written int64
+	)
 
-	iReader := itemData.ToReader()
-	progReader, closer := observe.ItemProgress(
-		ctx,
-		iReader,
-		observe.ItemRestoreMsg,
-		clues.Hide(name),
-		ss.Size())
+	// This is just to retry file upload, the uploadSession creation is not retried here
+	// We need extra logic to retry file upload as we have to pull the file again from kopia
+	for i := 0; i <= maxUploadRetries; i++ {
```

Comment:
not really sure where to put this comment so adding here...

This loop needs clearer definitions of what happens if there's an error. Do we attempt to upload all of the file data again? Do we begin uploading where we left off or near where we left off (maybe reupload the last `copyBufferSize` set of bytes as well as we may not know their state)

It's fine if we don't go for the super optimized solution right now, but we should make it clear what the behavior is. If we want to optimize it later we can track that with tickets

Reply:
It currently restarts the upload from beginning. I've created an issue (https://github.com/alcionai/corso/issues/3521) to track resuming upload.